File: Templates/Pyton/gameElements.py
import json


class FleetCommand:
    def __init__(self, sourcePlanetId, destinationPlanetId, numberOfUnits):
        self.SourcePlanetId = sourcePlanetId
        self.DestinationPlanetId = destinationPlanetId
        self.NumberOfUnits = numberOfUnits

    def toJSON(self):
        return json.dumps(self.__dict__)


class Bot:
    def __init__(self, id, name):
        self.Id = id
        self.Name = name
    
class Position:
    def __init__(self, x, y):
        self.X = x
        self.Y = y

class Planet:
    def __init__(self, id, ownerId, position, numberOfShips, growthRate):
        self.Id = id
        self.OwnerId = ownerId
        self.Position = Position(position["x"], position["y"])
        self.NumberOfShips = numberOfShips
        self.GrowthRate = growthRate

class Fleet:
    def __init__(self, id, ownerId, sourcePlanetId, destinationPlanetId, numberOfShips, ticksToDestination, position):
        self.Id = id
        self.OwnerId = ownerId
        self.NumberOfShips = numberOfShips
        self.TicksToDestination = ticksToDestination
        self.DestinationPlanetId = destinationPlanetId
        self.SourcePlanetId = sourcePlanetId
        self.Position = Position(position["x"], position["y"])

class GameState:
    def __init__(self, bots, planets, fleets):
        self.Bots=[]
        for bot in bots:    
            self.Bots.append(Bot(   bot["id"], 
                                    bot["name"]))

        self.Planets=[]
        for planet in planets:    
            self.Planets.append(Planet( planet["id"], 
                                        planet["ownerID"], 
                                        planet["position"], 
                                        planet["numberOfShips"],
                                        planet["growthRate"]))
        
        self.Fleets=[]
        for fleet in fleets:    
            self.Fleets.append(Fleet(   fleet["id"], 
                                        fleet["ownerID"], 
                                        fleet["sourcePlanetID"], 
                                        fleet["destinationPlanetID"], 
                                        fleet["numberOfShips"], 
                                        fleet["ticksToDestination"], 
                                        fleet["position"]))

File: Testing/GameElements.cs
using System.Collections.Generic;

namespace BattleBotTester
{
	public class GameElements
	{
		public class Bot
		{
			public int Id { get; set; }
			public string Name { get; set; }
		}

		public class Position
		{
			public int X { get; set; }
			public int Y { get; set; }
		}

		public class Planet
		{
			public int Id { get; set; }
			public int OwnerId { get; set; }
			public Position Position { get; set; }
			public int NumberOfShips { get; set; }
			public int GrowthRate { get; set; }
		}

		public class Fleet
		{
			public int Id { get; set; }
			public int OwnerId { get; set; }
			public int NumberOfShips { get; set; }
			public int TicksToDestination { get; set; }
			public int DestinationPlanetId { get; set; }
			public int SourcePlanetId { get; set; }
			public Position Position { get; set; }
		}

		public class GameState
		{
			public List<Bot> Bots { get; set; }
			public List<Planet> Planets { get; set; }
			public List<Fleet> Fleets { get; set; }
		}

		public class FleetCommand
		{
			public int SourcePlanetId { get; set; }
			public int DestinationPlanetId { get; set; }
			public int NumberOfUnits { get; set; }

			public FleetCommand(int sourcePlanetId, int destinationPlanetId, int numberOfUnits)
			{
				SourcePlanetId = sourcePlanetId;
				DestinationPlanetId = destinationPlanetId;
				NumberOfUnits = numberOfUnits;
			}
		}
	}
}